### PR TITLE
qrupdate: ILP64 support

### PR DESCRIPTION
**Pull request by siko1056**
- When compiling qrupdate with `FFLAGS=-fdefault-integer-8` it can be perfectly used for larger problem dimensions.
- Improved the readability of the file with the added rules.

### CODE REVIEW

File: var/spack/repos/builtin/packages/qrupdate/package.py [hunk 1]
```diff
@@ -29,17 +29,31 @@ def edit(self, spec, prefix):
...
+        make_args = [
+            'BLAS={0}'.format(lapack_blas.ld_flags),
+            'LAPACK={0}'.format(lapack_blas.ld_flags)
+        ]
+
+        # If 64-bit BLAS is used:
+        if (spec.satisfies('^openblas+ilp64') or
+            spec.satisfies('^intel-mkl+ilp64') or
+            spec.satisfies('^intel-parallel-studio+mkl+ilp64')):
+            make_args.append('FFLAGS=-fdefault-integer-8')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/qrupdate/package.py]: Wondering if this should be an option or attribute exposed by `blas` providers. In that way one could write something similar to:
```python
if spec['blas'].ilp64:
    make_args.append('FFLAGS=-fdefault-integer-8')

```

**Comment by adamjstewart** [comment]
@alalazo No opinions from me

File: var/spack/repos/builtin/packages/qrupdate/package.py (hunk 1 above)
Comment (siko1056) [reply var/spack/repos/builtin/packages/qrupdate/package.py]: @alalazo  This feature would indeed be nice. Should I open a separate issue for it and this patch gets applied as is for now?

Comment (alalazo) [reply var/spack/repos/builtin/packages/qrupdate/package.py]: Fair enough. If you could open an issue and ping me back for merge it would be great!

**Comment by siko1056** [comment]
Thank you @alalazo :+1:

**Pull request merged**